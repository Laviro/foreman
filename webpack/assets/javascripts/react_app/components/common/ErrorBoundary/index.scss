.error-boundary-foreman-app {
  .empty-state-icon {
    color: var(--pf-global--palette--red-200);
  }
}
