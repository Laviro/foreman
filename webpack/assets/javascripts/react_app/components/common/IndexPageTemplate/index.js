import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Table from '../../PF4/Table';
import PageLayout from '../../../routes/common/PageLayout/PageLayout';
import { fetchData } from '../../PF4/Table/TableActions';
import { getControllerSearchProps } from '../../../constants';
import './index.scss';
import { selectSearch, selectResponse } from '../../PF4/Table/TableSelectors';
import { getTableAPIKey } from '../../PF4/Table/TableHelpers';

const IndexPageTemplate = ({
  path,
  layoutProps,
  tableProps,
  paginationProps,
}) => {
  const dispatch = useDispatch();
  const searchQuery = useSelector(selectSearch);
  const { title, header } = useSelector(state =>
    selectResponse(state, getTableAPIKey(path))
  );
  const onSearch = nextQuery =>
    dispatch(fetchData(path, { query: nextQuery, page: 1 }));
  return (
    <PageLayout
      className="index_page_template"
      searchable
      searchQuery={searchQuery}
      searchProps={getControllerSearchProps(path)}
      onSearch={onSearch}
      header={header}
      title={title}
      {...layoutProps}
    >
      <Table
        requestPath={path}
        {...tableProps}
        paginationProps={paginationProps}
      />
    </PageLayout>
  );
};

export default IndexPageTemplate;
