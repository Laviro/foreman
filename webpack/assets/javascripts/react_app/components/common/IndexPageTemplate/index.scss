#main.index_page_template {
  padding: 0;

  h1 {
    font-weight: bold;
    margin-bottom: 0;
  }

  #index-page-top-pagination {
    margin-right: 24px;
  }

  .index-page-description {
    white-space: nowrap;
    margin-bottom: 15px;
  }
}
