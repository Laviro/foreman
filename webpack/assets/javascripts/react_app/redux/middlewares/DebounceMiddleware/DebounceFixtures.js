import { DEFAULT_DEBOUNCE } from './DebounceConstants';
import { withDebounce } from './DebounceHelpers';

export const type = 'SOME_TYPE';
export const key = 'SOME_KEY';
export const payload = { test: true };
export const debounce = DEFAULT_DEBOUNCE;
export const debounceID = 121;
export const initialState = {};
export const stateWithKey = { [key]: debounceID };
export const action = { type, key, payload };
export const actionWithDebounce = withDebounce(action);
