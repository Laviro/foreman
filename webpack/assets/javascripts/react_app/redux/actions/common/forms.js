import { APIActions } from '../../API';

import { addToast } from '../toasts';
import { sprintf, translate as __ } from '../../../../react_app/common/I18n';

export class SubmissionError {
  constructor(errors) {
    this.errors = errors;
  }
}

const fieldErrors = ({ error }) => {
  const { errors, severity } = error;

  if (errors.base) {
    errors._error = {};
    errors._error.errorMsgs = errors.base;
    errors._error.severity = severity;
    delete errors.base;
  }

  return new SubmissionError(errors);
};

export const onError = error => {
  if (error.response.status === 422) {
    // Handle invalid form data
    throw fieldErrors(error.response.data);
  }
  throw new SubmissionError({
    _error: {
      errorMsgs: [
        `${__('Error submitting data:')} ${error.response.status} ${__(
          error.response.statusText
        )}`,
      ],
    },
  });
};

const verifyProps = (item, values) => {
  if (!item) {
    throw new Error('item must be defined, e.g. Bookmark');
  }
  if (!values) {
    throw new Error('values must be defined');
  }
};

export const submitForm = ({
  item,
  url,
  values: params,
  message,
  method = 'post',
  headers,
  apiActionTypes: actionTypes,
}) => {
  verifyProps(item, params);
  return async dispatch => {
    const uniqueAPIKey = `${item.toUpperCase()}_FORM_SUBMITTED`;

    const handleError = error => onError(error);

    const handleSuccess = ({ data }) => {
      dispatch({
        type: uniqueAPIKey,
        payload: { item, data },
      });
      dispatch(
        addToast({
          type: 'success',
          // eslint-disable-next-line no-undef
          message: message || sprintf('%s was successfully created.', __(item)),
        })
      );
    };

    dispatch(
      APIActions[method]({
        key: uniqueAPIKey,
        url,
        headers,
        params,
        actionTypes,
        handleError,
        handleSuccess,
      })
    );
  };
};
