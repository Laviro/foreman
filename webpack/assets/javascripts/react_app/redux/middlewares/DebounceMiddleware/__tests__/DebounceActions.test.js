import { testActionSnapshotWithFixtures } from '../../../../common/testHelpers';
import { key, debounceID, debounce, timestamp } from '../DebounceFixtures';
import {
  clearDebounce,
  startDebounce,
  stopIncomingAction,
} from '../DebounceActions';

const fixtures = {
  'should start debounce': () =>
    startDebounce({ key, debounce, debounceID, timestamp }),
  'should clear debounce': () => clearDebounce(key),
  'should inform about a canceled incoming action while debounce': () =>
    stopIncomingAction(key),
};

describe('debounce actions', () => testActionSnapshotWithFixtures(fixtures));
