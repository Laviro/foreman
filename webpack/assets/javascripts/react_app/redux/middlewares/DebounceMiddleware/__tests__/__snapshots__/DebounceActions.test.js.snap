// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debounce actions should clear debounce 1`] = `
Object {
  "payload": Object {
    "key": "SOME_KEY",
  },
  "type": "DEBOUNCE_CLEAR",
}
`;

exports[`debounce actions should inform about a canceled incoming action while debounce 1`] = `
Object {
  "payload": Object {
    "key": "SOME_KEY",
  },
  "type": "DEBOUNCE_STOP_INCOMING_ACTION",
}
`;

exports[`debounce actions should start debounce 1`] = `
Object {
  "payload": Object {
    "debounceID": 121,
    "key": "SOME_KEY",
  },
  "type": "DEBOUNCE_START",
}
`;
